### ISSUE TITLE
Add projectDir option for WebpackConfigFactory.create

### ISSUE THREAD

**Issue by oreofeolurin** [nestjs/ng-universal#49]
## I'm submitting a...

<pre><code>
[ ] Regression 
[ ] Bug report
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

When using a different angular project structures like Nx Workspace, the `src` folder could end up being `apps/todo/src`.

`ng-univeral` `WepbackConfigFactory` configues the `ContextReplacementPlugin` to defaults to `src` which throws an error in a different angular project.

https://github.com/nestjs/ng-universal/blob/f4bb9f011d4b2626afbd16f0e94f1d375e1f7611/lib/webpack-config.factory.ts#L49

## Expected behavior

The `WebpackConfigFactory` static method `create` should recieve a `projectDir` option to properly configure the `ContextReplacementPlugin`

## What is the motivation / use case for changing the behavior?
Ability to use `ng-universal` with other project structures like `nx-workspace`

## Environment

<pre><code>
Nest version: X.Y.Z

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

**Comment by oreofeolurin** [comment]
A proposed solution looks more like this.

```
export class WebpackConfigFactory {
  static create(
    webpack: any,
    entry: WebpackEntries = defaultEntries,
    currentDir: string = process.cwd(),
    projectDir: string = currentDir
  ) {
```

`projectDir`  can be used like this
```
plugins: [
        new webpack.ContextReplacementPlugin(
          // fixes WARNING Critical dependency: the request of a dependency is an expression
          /((.+)?angular(\\|\/)core(.+)?|express(.+)?|(.+)?nestjs(\\|\/)(.+)?)?/,
          join(projectDir, 'src'), // location of your src
          {}
        )
      ]
```

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR?

**Comment by oreofeolurin** [comment]
Yes @kamilmysliwiec, would send one right away.

**Comment by kamilmysliwiec** [comment]
Published as 1.1.0
